File: src/main/java/classes/MainWindow.java
package classes;

import javax.swing.*;
import java.awt.*;

public class MainWindow extends JFrame {
    private GameField gameField;
    private Dimension fieldSize;

    public MainWindow() {
        this.setTitle("Змейка");
        this.setDefaultCloseOperation(WindowConstants.EXIT_ON_CLOSE);

        // Полноэкранный режим
        this.setExtendedState(JFrame.MAXIMIZED_BOTH);
        this.setUndecorated(true);

        this.fieldSize = Toolkit.getDefaultToolkit().getScreenSize();
        this.gameField = new GameField(this.fieldSize);

        this.add(this.gameField);
    }

    public void close() {
        this.removeAll();
        this.dispose();
    }
}

File: src/main/java/classes/MainWindow.kt
package classes

import java.awt.Dimension
import java.awt.Toolkit
import javax.swing.JFrame

class MainWindow : JFrame() {
    private val gameField: GameField
    private val fieldSize: Dimension

    init {
        title = "Змейка"
        defaultCloseOperation = EXIT_ON_CLOSE

        // Полноэкранный режим
        this.extendedState = MAXIMIZED_BOTH
        this.isUndecorated = true
        fieldSize = Toolkit.getDefaultToolkit().screenSize
        gameField = GameField(fieldSize)
        this.add(gameField)
    }

    fun close() {
        this.removeAll()
        dispose()
    }
}